src/diffusers/models/vae.py

```diff
@@ -337,12 +337,16 @@ def __init__(self, parameters, deterministic=False):
         self.std = torch.exp(0.5 * self.logvar)
         self.var = torch.exp(self.logvar)
         if self.deterministic:
-            self.var = self.std = torch.zeros_like(self.mean).to(device=self.parameters.device)
+            self.var = self.std = torch.zeros_like(self.mean).to(
+                device=self.parameters.device, dtype=self.parameters.dtype
+            )
 
     def sample(self, generator: Optional[torch.Generator] = None) -> torch.FloatTensor:
         device = self.parameters.device
         sample_device = "cpu" if device.type == "mps" else device
-        sample = torch.randn(self.mean.shape, generator=generator, device=sample_device).to(device)
+        sample = torch.randn(self.mean.shape, generator=generator, device=sample_device).to(
+            device=device, dtype=self.parameters.dtype
+        )
```

Comment:
Same as previous comment

Reply:
no here it's fine for gpu/cpu reproducability

Reply:
Sorry im not aware of this. Does having the following create reproducability issues? Can you point me to a related PR/issue if it's the case?
```python
torch.randn(self.mean.shape, generator=generator, device=sample_device, dtype=self.parameters.dtype)
```

Reply:
I believe when the generator is not None, doing randn on the GPU with the same generator will give different results than on the CPU and so we currently make the noise etc. on the CPU with the generator and then move it to the device...  see for example https://github.com/huggingface/diffusers/blob/main/src/diffusers/schedulers/scheduling_ddim.py#L287

Reply:
Thanks for the explanation 🙏🏼